### PR TITLE
Implemented Statement::fetch(PDO::FETCH_COLUMN) in non-PDO driveres

### LINKED ISSUE #2953: Unlike other drivers, DBAL has a hard dependency on PDO
Historically, the DBAL public API uses `PDO` constants which creates a hard dependency and makes the abstraction not that abstract. For the end user, it means that even if none of the `PDO` drivers are going to be used, the production environment still has to have `ext-pdo` installed.

To eliminate the dependency, the `PDO` constants in the method signatures should be replaced by the DBAL's own ones:

1. `PDO::PARAM_*` → `Doctrine\DBAL\Driver\Statement::PARAM_*`,
2. `PDO::FETCH_*` → `Doctrine\DBAL\Driver\ResultStatement::FETCH_*`,
3. `PDO::CASE_*` → `Doctrine\DBAL\Driver\ResultStatement::CASE_*`.

It can be done in phases:

1. In `2.x`, introduce the new DBAL constants and assign them corresponding PDO values, e.g. `Statement::PARAM_INT = PDO::PARAM_INT`.
2. Deprecate the usage of `PDO::` constants when calling the DBAL methods.
3. In `3.0`, replace the `PDO` constants with their integer values, e.g. `Statement::PARAM_INT = 1`
4. Remove the dependency.

### LINKED ISSUE #2963: OCI8Statement::fetch() ignores the PDO::FETCH_COLUMN mode
The issue was discovered by a [continuousphp build](https://continuousphp.com/git-hub/doctrine/dbal/build/fea37953-01cc-4297-b23a-8de4b06d3e06). It could be reproduced as:
```php
$conn = \Doctrine\DBAL\DriverManager::getConnection([
    'driver' => 'oci8',
    'host' => 'localhost',
    'user' => 'system',
    'password' => 'oracle',
    'dbname' => 'XE',
]);

$foo = $conn->executeQuery('SELECT \'foo\' FROM DUAL')->fetch(PDO::FETCH_COLUMN);
var_dump($foo);

/*
expected: string(3) "foo"
actual: array(1) {
  [0] =>
  string(3) "foo"
}
*/
```

### PR DESCRIPTION
**Pull request by morozov**
Fixes #2963.

### CODE REVIEW

File: lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php [hunk 1]
```diff
@@ -272,6 +272,12 @@ public function fetch($fetchMode = null, $cursorOrientation = \PDO::FETCH_ORI_NE
             return false;
         }
 
+        $fetchMode = $fetchMode ?: $this->_defaultFetchMode;
+
+        if ($fetchMode == PDO::FETCH_COLUMN) {
```
Comment (Majkl578) [inline lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php]: Could we use `===` here (and below)?

File: tests/Doctrine/Tests/DBAL/Functional/StatementTest.php [hunk 2]
```diff
@@ -282,4 +282,13 @@ function (Statement $stmt) {
             ),
         );
     }
+
+    public function testFetchInColumnMode()
```
Comment (Majkl578) [inline tests/Doctrine/Tests/DBAL/Functional/StatementTest.php]: ` : void`

**Review by Majkl578 — APPROVED** [review]

**Comment by Ocramius** [comment]
LGTM 👍

**Comment by Ocramius** [comment]
@morozov does this need a backport to 2.6.x?

**Comment by morozov** [comment]
@Ocramius, yes.

**Comment by Ocramius** [comment]
@lcobucci was this already backported? I didn't do that, so the milestone change may be incorrect until it's done.

**Pull request merged**